### DISCUSSION TITLE
Git add .

### DISCUSSION THREAD

**Discussion by strikeouts27** [community/community#64999] [Programming Help]
### Body

I tried to go back to a previous commit and start my project over, but I was always running into errors and it has been a whole day of mispent time. 

Backstory here: https://stackoverflow.com/questions/76981288/trying-to-go-back-to-previous-commits-in-github-while-accidently-uploading-anoth?noredirect=1#comment135708264_76981288

So I finally decided to grab the most recent file and download the zip files. After doing that, I deleted the repository and created a new one and uploaded the files. When I attempt to do so I run git status and I get this output. I have attempted to do command shift p . to see the hidden files but I have been unable to do so. 
When I type in git add . I git this monster list of outputs. Is there a way to unstage them all at once or do I have to go in line by line? How did these ../ entries get there? 

I was running git add . but that was from my project folder. How did these other files get into the staging area? I have run git reset and I have NOT made any commits to the github repository yet. 

`37 files changed, 797 insertions(+)
 create mode 100644 Desktop/news/.gitignore
 create mode 100644 Desktop/news/.vscode/settings.json
 create mode 100644 Desktop/news/README.MD
 create mode 100644 Desktop/news/accounts/__init__.py
 create mode 100644 Desktop/news/accounts/admin.py
 create mode 100644 Desktop/news/accounts/apps.py
 create mode 100644 Desktop/news/accounts/forms.py
 create mode 100644 Desktop/news/accounts/migrations/0001_initial.py
 create mode 100644 Desktop/news/accounts/migrations/__init__.py
 create mode 100644 Desktop/news/accounts/models.py
 create mode 100644 Desktop/news/accounts/tests.py
 create mode 100644 Desktop/news/accounts/urls.py
 create mode 100644 Desktop/news/accounts/views.py
 create mode 100644 Desktop/news/django_project/__init__.py
 create mode 100644 Desktop/news/django_project/asgi.py
 create mode 100644 Desktop/news/django_project/settings.py
 create mode 100644 Desktop/news/django_project/urls.py
 create mode 100644 Desktop/news/django_project/wsgi.py
 create mode 100755 Desktop/news/manage.py
 create mode 100644 Desktop/news/pages/__init__.py
 create mode 100644 Desktop/news/pages/admin.py
 create mode 100644 Desktop/news/pages/apps.py
 create mode 100644 Desktop/news/pages/migrations/__init__.py
 create mode 100644 Desktop/news/pages/models.py
 create mode 100644 Desktop/news/pages/tests.py
 create mode 100644 Desktop/news/pages/urls.py
 create mode 100644 Desktop/news/pages/views.py
 create mode 100644 Desktop/news/templates/base.html
 create mode 100644 Desktop/news/templates/home.html
 create mode 100644 Desktop/news/templates/registration/login.html
 create mode 100644 Desktop/news/templates/registration/password_change_done.html
 create mode 100644 Desktop/news/templates/registration/password_change_form.html
 create mode 100644 Desktop/news/templates/registration/password_reset_complete.html
 create mode 100644 Desktop/news/templates/registration/password_reset_confirm.html
 create mode 100644 Desktop/news/templates/registration/password_reset_done.html
 create mode 100644 Desktop/news/templates/registration/password_reset_form.html
 create mode 100644 Desktop/news/templates/registration/signup.html
(base) andrewstribling@Andrews-MBP news % git add .
(base) andrewstribling@Andrews-MBP news % git commit -m "chapter 12"
On branch main
Your branch is ahead of 'origin/main' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
        ../../.CFUserTextEncoding
        ../../.DS_Store
        ../../.Trash/
        ../../.anaconda/
        ../../.bash_history
        ../../.bash_profile
        ../../.bashrc.save
        ../../.conda/
        ../../.condarc
        ../../.config/
        ../../.continuum/
        ../../.cups/
        ../../.docker/
        ../../.fly/
        ../../.gitconfig
        ../../.idlerc/
        ../../.ipynb_checkpoints/
        ../../.ipython/
        ../../.jupyter/
        ../../.lesshst
        ../../.matplotlib/
        ../../.mono/
        ../../.npm/
        ../../.pgadmin/
        ../../.psql_history
        ../../.python_history
        ../../.ssh/
        ../../.tcshrc
        ../../.viminfo
        ../../.vscode/argv.json
        ../../.vscode/cli/
        ../../.vscode/extensions/
        ../../.xonshrc
        ../../.zprofile
        ../../.zsh_history
        ../../.zsh_sessions/
        ../../.zshrc
        ../../Andrews_First_Jupyter_Notebook.ipynb
        ../../Applications/
        ../../Baseball_Databank/
        ../../Codecademy_Resturant.sql
        ../../DJANGO_Meetup/
        ../.DS_Store
        ../.localized
        ../.ssh:config
        ../6 Popular Data Analytics Certifications_ Your 2023 Guide _ Coursera_files/
        ../Andrew Stribling Pathways.docx
        ../Codecademy learn Data analyst.  copy.pdf
        ../ID THEFT AFFIDAVIT FTC.pdf
        ../Programming/
        ../Screen Shot 2023-08-21 at 10.49.51 PM.png
        ../convert_csv/
        ../keyboard-shortcuts-macos.pdf
        ../kobe-angry-949098398.jpg
        ../../Django_Tutorial_Polls_App
        ../../Documents/
        ../../Downloads/
        ../../Library/
        ../../Movies/
        ../../Music/
        ../../Pictures/
        ../../Pipfile
        ../../Pipfile.lock
        ../../Public/
        ../../Untitled.ipynb
        ../../anaconda3/
        ../../andrew-board/
        ../../django_for_beginners/
        ../../djangodelights/
        ../../get-pip.py
        ../../gitignore
        ../../jupyter.py
        ../../mysite/
        ../../placeholder.py
        ../../project_env/
        ../../python.py
        ../../untitled.txt
        ../../venv/
        ../../welcome-to-docker/`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by proksh** [comment]
From the logs you provided, it seems that when you ran the git add . command, you added not only the files in your project directory (news) but also many unrelated files and directories from your home folder and parent directories. This probably happened because you initialized a Git repository at a higher level directory than you intended to.

To solve this, you should take the following steps:

1. **Reset the Staging Area**: This will unstage all the changes.
```
git reset
```

2. **Ensure You're in the Correct Directory**: Make sure you are in the root directory of your project when working with Git commands. In your case, this seems to be the `news` folder.

3. **Initialize Git (If Needed)**: If you haven't initialized a Git repository in your project directory, do so:
```
cd /path/to/your/project/news
git init
```

4. **Add and Commit Your Files Again**: Now, when you're sure you are in the correct directory, run:
```
git add .
git commit -m "Your commit message"
```

5. **Check the Status**: To ensure you've added the right files, always run:
```
git status
```

6. **Add a .gitignore File (If Missing)**: If there are files or directories that you don't want to track, such as environment files, cached files, etc., you can list them in a `.gitignore` file in the root of your project. This will prevent these files from being tracked by Git.

7. **Double-check Before Pushing**: Always ensure you review the files you're about to push. This can save a lot of potential headaches.

A note for future projects: Always make sure to initialize your Git repository in the root of your project to avoid accidentally tracking unrelated files.

**Reply by strikeouts27** [reply]
It seems to want to initalize a repository at .git based on the following out put of git init.
Initialized empty Git repository in /Users/andrewstribling/Desktop/Programming/projects/django/news/.git/

I am pretty certain I do not want to start my repository there but instead start it in the news folder. How do I get it to stop initializing in  the .git file?

**Reply by GhostOf0days** [reply]
It will be initialized with a new folder in the folder you are in. Use cd .. to move back into the desired folder (i.e. projects), a new folder should show up if I remember correctly. Move the project files into that folder.
